test/modules/standard/gmp/ferguson/gmp_dist_array.chpl

```diff
@@ -44,16 +41,27 @@ modulus.nextprime(modulus);
 writeln(modulus);
 sum.mod(sum, modulus);
 writeln(sum);
-delete modulus;
-delete sum;
+
 
 // Release the array elements.
+// TODO: remove this section once the records are being released correctly.
```

Comment:
It’s well known that distributed arrays leak their data and there is an
existing project to tackle this.  We won’t want to add memory management
to other types to work around the current problems issues in distributed arrays.

Mike



On Jul 28, 2016, at 11:59 AM, Laura Delaney <notifications@github.com<mailto:notifications@github.com>> wrote:


In test/modules/standard/gmp/ferguson/gmp_dist_array.chpl<https://github.com/chapel-lang/chapel/pull/4243#discussion_r72681703>:

>
>  // Release the array elements.
> +// TODO: remove this section once the records are being released correctly.


the distributed array leaks all of its array elements. Since the elements are leaked, the records don't go out of scope and their destructors aren't called. Manually calling the destructors significantly reduces the amount of memory leaked.

Ideally, it wouldn't need to think about the memory management, and ideally it wouldn't be leaking the array elements. In the meantime, manually freeing the records means that we leak around 2000x less data from this one test.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub<https://github.com/chapel-lang/chapel/pull/4243/files/3bd5d074a99d2427218177f58af6019dc3d8c2be#r72681703>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ACERpqyEnggd1aIq0yYCU3VC3tpxWGw_ks5qaPwJgaJpZM4JTwJ0>.